File: Data Structures & Algorithms/pow-x-n/submission-0.java
class Solution {
    public double myPow(double x, int n) {
        double res = helper(x, Math.abs(n));
        return n < 0 ? 1/res : res;
    }

    private double helper(double x, int n){
        if(x == 0){
            return 0;
        }

        if(n == 0){
            return 1;
        }

        double res = helper(x, n/2);
        res *= res;
        return n % 2 != 0 ? x * res : res;
    }
}

File: Data Structures & Algorithms/extra-characters-in-a-string/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @param {string[]} dictionary
     * @return {number}
     */

    // neetcodes

    minExtraChar(s, dictionary) {
        const backtrack = (i) => {
            if(i == s.length) return 0;

            let res = 1 + backtrack(i + 1);

            for(let w of dictionary){
                const j = w.length;

                if(s.slice(i, j) === w){
                    res = Math.min(res, backtrack(j));
                }
            }

            return res;
        }

        return backtrack(0);
    }
}
